cam_diagnostics/cam_diagnostics.F90

```diff
@@ -0,0 +1,211 @@
+module cam_diagnostics
+
+   use ccpp_kinds, only:  kind_phys
+   use ccpp_constituent_prop_mod, only: ccpp_constituent_prop_ptr_t
+
+   implicit none
+   private
+   save
+
+   public :: cam_state_diagnostics_init ! init routine
+   public :: cam_state_diagnostics_run  ! main routine
+   public :: cam_tend_diagnostics_init ! init routine
+   public :: cam_tend_diagnostics_run  ! main routine
+
+   character(len=65) :: const_std_names(4) = &
+   (/'water_vapor_mixing_ratio_wrt_moist_air_and_condensed_water       ', &
+     'cloud_liquid_water_mixing_ratio_wrt_moist_air_and_condensed_water', &
+     'rain_mixing_ratio_wrt_moist_air_and_condensed_water              ', &
+     'cloud_ice_mixing_ratio_wrt_moist_air_and_condensed_water         '/)
+
+   character(len=6) :: const_diag_names(4) = (/'Q     ', &
+                                               'CLDLIQ', &
+                                               'CLDICE', &
+                                               'RAINQM'/)
+
+CONTAINS
+
+   !> \section arg_table_cam_state_diagnostics_init  Argument Table
+   !! \htmlinclude cam_state_diagnostics_init.html
+   subroutine cam_state_diagnostics_init(const_props, errmsg, errflg)
+      use cam_history,         only: history_add_field
+      use cam_history_support, only: horiz_only
+
+      type(ccpp_constituent_prop_ptr_t), intent(in) :: const_props(:)
+      character(len=512), intent(out) :: errmsg
+      integer,            intent(out) :: errflg
+
+      ! Local variables:
+
+      integer :: const_idx, name_idx
+      integer :: const_num_found
+      character(len=512) :: standard_name
+
+      errmsg = ''
+      errflg = 0
+
+      ! Add state fields
```

Comment:
What about the fields like te_ini, te_cur, tw_ini, tw_cur, temp_ini, z_ini?  These are all fields that are in CAM's current state, but not included in your list

Reply:
made issue https://github.com/ESCOMP/atmospheric_physics/issues/128